tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/PopulateBasicPropertiesHeadersIntegration.cs

```diff
@@ -31,6 +31,12 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ;
 [EditorBrowsable(EditorBrowsableState.Never)]
 public class PopulateBasicPropertiesHeadersIntegration
 {
+    internal static CallTargetState OnMethodBegin<TTarget, TBasicProperties, TActivity>(TTarget instance, TBasicProperties basicProperties, TActivity sendActivity, ulong publishSequenceNumber)
+        where TBasicProperties : IReadOnlyBasicProperties, IDuckType
```

Comment:
using the constraint-based version creates a struct ducktype, which is _normally_ the best option, but then we're immediately boxing it as an `object` by passing it as `state` so I'm not sure if that's what we want to do or not tbh 🤔 I would normally ask @tonyredondo for his opinion at this point 😄 

Given that there's a chance we _don't_ need to duck type it (e.g. because `returnValue` is non-null) I'd suggest _not_ using the constraint duck typing here, and instead passing the object itself, and doing the duck typing in OnMethodEnd instead

Reply:
I agree if we are not using the ducktype from the constraint and we are just boxing it then I prefer to not doing the constraint at all just pass the original instance to the state.

Reply:
Done.